src/transformers/trainer.py

```diff
@@ -1875,17 +1877,40 @@ def train(
 
         # Load potential model checkpoint
         if isinstance(resume_from_checkpoint, bool) and resume_from_checkpoint:
-            resume_from_checkpoint = get_last_checkpoint(args.output_dir)
-            if resume_from_checkpoint is None:
+            checkpoints = get_checkpoints(args.output_dir, reverse=True)
+            if not checkpoints:
                 raise ValueError(f"No valid checkpoint found in output directory ({args.output_dir})")
+        else:
+            checkpoints = [resume_from_checkpoint] if resume_from_checkpoint else []
 
-        if resume_from_checkpoint is not None:
+        if checkpoints:
             if not is_sagemaker_mp_enabled() and not self.is_deepspeed_enabled and not self.is_fsdp_enabled:
-                self._load_from_checkpoint(resume_from_checkpoint)
+                for i, checkpoint in enumerate(checkpoints):
+                    try:
+                        self._load_from_checkpoint(checkpoint)
+                        if i > 0:
+                            logger.info(f"Checkpoints {checkpoints[i:]} failed to load. Loading from {checkpoint}.")
+                        checkpoints = [checkpoint]
+                        break
+                    except (ValueError, safetensors.SafetensorError) as e:
+                        logger.warning(f"Error loading checkpoint {checkpoint}: {e}")
+                else:
+                    logger.warning(f"No valid checkpoint found in output directory ({args.output_dir}), starting from scratch")
+                    checkpoints = []
```

Comment:
Let's fully tell the user they don't have valid checkpoints, rather than keep going
```suggestion
                for i, checkpoint in enumerate(checkpoints):
                    try:
                        self._load_from_checkpoint(checkpoint)
                        checkpoints = [checkpoint]
                        break
                    except (ValueError, safetensors.SafetensorError) as e:
                        if i != len(checkpoints)-1:
                            logger.warning(f"Error loading checkpoint {checkpoint}: {e}\nMoving onto the next checkpoint")
                else:
                    raise ValueError(f"No valid checkpoints were found in output directory {args.output_dir}")
```

Reply:
Do you mean just try the last (or specified one) and raise an error ?